juniper_junos-formula/bin/compile_junos_data.py

```diff
@@ -0,0 +1,371 @@
+#!/usr/bin/python3
+"""
+Juniper Junos Salt pillar generator
+Copyright (C) 2023 SUSE LLC <georg.pfuetzenreuter@suse.com>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+import argparse
+import logging
+import sys
+import yaml
+
+logger = logging.getLogger(__name__)
+log_choices_converter = {
+        'debug': logging.DEBUG,
+        'info': logging.INFO
+        }
+argparser = argparse.ArgumentParser()
+argparser.add_argument('--log', type=str, default='info', choices=log_choices_converter.keys())
+argparser.add_argument('--in-switching', type=str, default='switching.yaml')
+argparser.add_argument('--in-backbone', type=str, default='backbone.yaml')
+argparser.add_argument('--out', type=str, default='output')
+args = argparser.parse_args()
+
+infile_s = args.in_switching
+infile_b = args.in_backbone
+outdir = args.out
+
+def generate_switch_pillars(data):
```

Comment:
Fair point, and having some individual functions which could potentially be imported to other scripts could be useful. Unfortunately I don't have time to refactor it further and to investigate possible regressions (would be nice to have a test case for the script as well ..) at the moment. Can we keep the idea as a "nice to have" for the future?

Reply:
Experience shows that "nice to have for the future" translates to "never" ;-) - and "TODO" notes are not really better.

Of course I can't force you to do anything, but I'd still love to be proven wrong with the above statement *g*

Reply:
If nobody steps up to contribute this function split, then yes, it will likely be never. I tried to get the most out of this first refactor but there is still lots of work to do.

Reply:
Can we keep it?

Reply:
I'd love to say "no" - but since you don't have time to do the split now, that would not be too helpful ;-) and I don't want block this PR because of this.

However, adding a TODO note would be nice ;-)

Reply:
Already done, as discussed! Appreciated!